lib/src/signed_video_h26x_auth.c

```diff
@@ -211,23 +210,14 @@ prepare_for_link_and_gop_hash_verification(signed_video_t *self, h26x_nalu_list_
 
       // Skip GOP SEI items as they do not contribute to the GOP hash.
       if (item->nalu->is_gop_sei) {
-        item = item->next;
-        if (item) {
-          // Track when the current item follows the SEI or marks the beginning of the next GOP.
-          found_item_after_sei = (item->prev == sei);
-        } else {
-          found_next_gop = true;
-        }
-        continue;
+        break;  // Break if encountered SEI frame.
```

Comment:
Is this really safe? Shouldn't you break at ANY sei but rather THE sei which equals the input?